internal/providers/terraform/aws/kinesis_firehose_delivery_stream.go

```diff
@@ -0,0 +1,131 @@
+	costComponents := make([]*schema.CostComponent, 0)
+	var monthlyDataIngestedGb *decimal.Decimal
+	var requestQuantities []decimal.Decimal
+	halfAMillion := []int{512000}
+	oneAndHalfPB := decimal.NewFromInt(1536000)
+
+	if u != nil && u.Get("monthly_data_ingested_gb").Type != gjson.Null {
+		monthlyDataIngestedGb = decimalPtr(decimal.NewFromInt(u.Get("monthly_data_ingested_gb").Int()))
+		requestQuantities = usage.CalculateTierBuckets(*monthlyDataIngestedGb, halfAMillion)
+		firstGB := requestQuantities[0]
+		nextOneAndHalfGB := requestQuantities[1]
+
+		if nextOneAndHalfGB.Equal(decimal.Zero) {
+
+			costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", &firstGB))
+		}
+		if nextOneAndHalfGB.GreaterThan(oneAndHalfPB) {
+			nextThreeGB := nextOneAndHalfGB.Sub(oneAndHalfPB)
+			costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", decimalPtr(decimal.NewFromInt(int64(halfAMillion[0])))))
+			costComponents = append(costComponents, kenesisFirehosCostComponent("next 1.5PB", region, "512000", "2048000", &oneAndHalfPB))
+			costComponents = append(costComponents, kenesisFirehosCostComponent("next 3PB", region, "2048000", "Inf", &nextThreeGB))
+		} else {
+			costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", decimalPtr(decimal.NewFromInt(int64(halfAMillion[0])))))
+			costComponents = append(costComponents, kenesisFirehosCostComponent("next 1.5PB", region, "512000", "2048000", &nextOneAndHalfGB))
+		}
+	} else {
+		var unknown *decimal.Decimal
+
+		costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", unknown))
+		costComponents = append(costComponents, kenesisFirehosCostComponent("next 1.5PB", region, "512000", "2048000", unknown))
+		costComponents = append(costComponents, kenesisFirehosCostComponent("next 3PB", region, "2048000", "Inf", unknown))
+	}
+
+	if d.Get("extended_s3_configuration.0.data_format_conversion_configuration.0.enabled").Type != gjson.Null {
+		enabled := d.Get("extended_s3_configuration.0.data_format_conversion_configuration.0.enabled")
+		if enabled.Type == gjson.True {
+			costComponents = append(costComponents, kenesisFirehosConversationCostComponent(region, monthlyDataIngestedGb))
+		}
+	}
+
+	if d.Get("elasticsearch_configuration").Type != gjson.Null {
+		elasticsearchConfiguration := d.Get("elasticsearch_configuration")
+		if elasticsearchConfiguration.Get("0.vpc_config").Type != gjson.Null {
+			costComponents = append(costComponents, kenesisFirehosVPCCostComponent("VPC data", region, "VpcBandwidth", monthlyDataIngestedGb))
+			vpcConfigs := elasticsearchConfiguration.Get("0.vpc_config")
+			zones := decimalPtr(decimal.NewFromInt(int64(len(vpcConfigs.Get("0.subnet_ids").Array()))))
+			costComponents = append(costComponents, kenesisFirehosVPCCostComponent("VPC AZ deilvery", region, "RunVpcInstance", zones))
```

Comment:
`VPC AZ deilvery` cost component should have `zones` as a unit, and used as `HourlyQuanitity`. It said in [pricing page](https://aws.amazon.com/kinesis/data-firehose/pricing/?nc=sn&loc=3): `Per hour, per AZ for VPC delivery`